pkg/types/types.go

```diff
@@ -72,6 +72,50 @@ var NodeRoles = map[string]Role{
 	string(LoadBalancerRole): LoadBalancerRole,
 }
 
+// ClusterHealth describes the health of a cluster
+type ClusterHealth string
+
+// possible health states
+const (
+	ClusterHealthOK               ClusterHealth = "OK"                  // all nodes running
+	ClusterHealthDegraded         ClusterHealth = "Degraded"            // 1+ nodes down
+	ClusterHealthOKModified       ClusterHealth = "OK (Modified)"       // some nodes stopped intentionally, rest is running
+	ClusterHealthDegradedModified ClusterHealth = "Degraded (Modified)" // 1+ nodes down and some more stopped intentionally
+	ClusterHealthBroken           ClusterHealth = "Broken"              // all nodes down
+)
+
+// ClusterHealthStates defines the possible health states for a cluster
+var ClusterHealthStates = map[string]ClusterHealth{
+	string(ClusterHealthOK):               ClusterHealthOK,
+	string(ClusterHealthDegraded):         ClusterHealthDegraded,
+	string(ClusterHealthOKModified):       ClusterHealthOKModified,
+	string(ClusterHealthDegradedModified): ClusterHealthDegradedModified,
+	string(ClusterHealthBroken):           ClusterHealthBroken,
+}
+
+// ClusterStatus defines a state that a cluster can be in
+type ClusterStatus string
+
+// possible cluster states
+const (
+	ClusterStatusUp      ClusterStatus = "Up"      // 1 or more servers running
+	ClusterStatusDown    ClusterStatus = "Down"    // no server running
+	ClusterStatusStopped ClusterStatus = "Stopped" // cluster stopped intentionally (k3d cluster stop mycluster)
```

Comment:
What about:
- Up :arrow_right: Running
- Down :arrow_right: Created
- Stopped :arrow_right: Stopped

or

- Up :arrow_right: Up
- Down :arrow_right: New
- Stopped :arrow_right: Down

Reply:
Status is about whether the cluster is usable (i.e. accessible via kubectl) or not, which is only about at least one server being up and running.
Changing `Down` to `New` or `Created` wouldn't implicate that the cluster crashed, would it? :thinking: